tests/codegen/paginator-tests/build.gradle.kts

```diff
@@ -19,82 +18,56 @@ buildscript {
 
 extra.set("skipPublish", true)
 
-val platforms = listOf("common", "jvm")
-
-platforms.forEach { platform ->
-    apply(from = rootProject.file("gradle/${platform}.gradle"))
-}
-
 val optinAnnotations = listOf("kotlin.RequiresOptIn", "aws.smithy.kotlin.runtime.util.InternalApi")
+kotlin.sourceSets.all {
+    optinAnnotations.forEach { languageSettings.optIn(it) }
+}
 
-kotlin {
-    sourceSets {
-        all {
-            val srcDir = if (name.endsWith("Main")) "src" else "test"
-            val resourcesPrefix = if (name.endsWith("Test")) "test-" else  ""
-            // the name is always the platform followed by a suffix of either "Main" or "Test" (e.g. jvmMain, commonTest, etc)
-            val platform = name.substring(0, name.length - 4)
-            kotlin.srcDir("$platform/$srcDir")
-            resources.srcDir("$platform/${resourcesPrefix}resources")
-            languageSettings.progressiveMode = true
-            optinAnnotations.forEach { languageSettings.optIn(it) }
-        }
-
-        val coroutinesVersion: String by project
-        val sdkVersion: String by project
-        commonMain {
-            dependencies {
-                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")
-                implementation(project(":runtime:runtime-core"))
-            }
-        }
-        val jvmMain by getting {
-            kotlin.srcDir("${project.buildDir}/generated-src/src")
-        }
-    }
+kotlin.sourceSets.getByName("main") {
+    kotlin.srcDir("${project.buildDir}/generated-src/src")
+    kotlin.srcDir("${project.buildDir}/smithyprojections/paginator-tests/paginator-tests/kotlin-codegen")
 }
 
-tasks["smithyBuildJar"].enabled = false
+val coroutinesVersion: String by project
+val sdkVersion: String by project
 
-val codegen by configurations.creating
+tasks["smithyBuildJar"].enabled = false
 
+val codegen: Configuration by configurations.creating
```

Comment:
you don't need this, it's an empty configuration. The reason it exists in [benchmarks](https://github.com/awslabs/smithy-kotlin/blob/main/tests/benchmarks/serde-benchmarks/build.gradle.kts#L88) is because we need to add a dependency that smithy build task requires to work the way we want.

You should be able to remove all references to it (e.g. L39 and L41)

Reply:
When I remove the code so that references to `codegen` are removed, I am left with:

```
tasks["smithyBuildJar"].enabled = false

val generateSdk = tasks.create<SmithyBuild>("generateSdk") {
    group = "codegen"
    inputs.file(projectDir.resolve("smithy-build.json"))
}

val stageGeneratedSources = tasks.register("stageGeneratedSources") {
    group = "codegen"
    dependsOn(generateSdk)
}
```

When I execute a test, I get the following error:

```
Projection paginator-tests failed: software.amazon.smithy.build.SmithyBuildException: Unable to find a plugin named `kotlin-codegen` in the `paginator-tests` projection. Is this the correct spelling? Are you missing a dependency? Is your classpath configured correctly?
software.amazon.smithy.build.SmithyBuildImpl.applyPlugin(SmithyBuildImpl.java:384)
software.amazon.smithy.build.SmithyBuildImpl.applyProjection(SmithyBuildImpl.java:325)
```

It's not clear to me what I would do other than keep the code as-is.  Suggestions?

Reply:
Yeah you need a dependency on `project(":smithy-kotlin-codegen")` and then either set `addRuntimeClasspath = true` or `addBuildScriptClasspath = true` in the `generateSdk` task depending on what you choose. It's unclear why this even works in the first place to be honest. 

Currently it looks like you have it defined in the normal [normal](https://github.com/awslabs/smithy-kotlin/blob/91d6d3bee6bef717626686af98d4d95fb3fbad34/tests/codegen/paginator-tests/build.gradle.kts#L69) dependencies block which would correspond to adding `addRuntimeClasspath`. 


Either that or leave `codegen` configuration there and change the dependencies block to look like:
```kotlin
dependencies {
    codegen(project(":smithy-kotlin-codegen"))
}
```

Reply:
thanks, that worked :+1: